typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,541 @@
+        if (!resultRegisterService.isOk()) {
+            return err(resultRegisterService.error);
+        }
+        const registerServiceResp: RegisterServiceResponse = resultRegisterService.value;
+
+        const serviceIpAddr = registerServiceResp.getIpAddr();
+
+        const serviceContext: ServiceContext = new ServiceContext(
+            this.client,
+            serviceId,
+            serviceIpAddr,
+            this.enclaveDataVolMountpoint,
+            containerCreationConfig.getKurtosisVolumeMountpoint());
+        log.trace("New service successfully registered with Kurtosis API");
+
+        log.trace("Loading static files into new service namespace...");
+        const usedStaticFilesMap = containerCreationConfig.getUsedStaticFiles();
+
+        const usedStaticFiles: Set<string> = new Set();
+        for (let usedStaticFilesId in usedStaticFilesMap) {
+            usedStaticFiles.add(usedStaticFilesId);
+        }
+
+        const resultLoadStaticFiles = await serviceContext.loadStaticFiles(usedStaticFiles); 
+        if (!resultLoadStaticFiles.isOk()) {
+            return err(resultLoadStaticFiles.error);
+        }
+        const staticFileAbsFilepathsOnService: Map<string, string> = resultLoadStaticFiles.value;
+        log.trace("Successfully loaded static files");
+
+        log.trace("Initializing generated files...");
+        const filesToGenerate: Set<string> = new Set();s
+        for (let fileId in containerCreationConfig.getFileGeneratingFuncs()) {
+            filesToGenerate.add(fileId);
+        }
+        const resultGenerateFiles = await serviceContext.generateFiles(filesToGenerate);
+        if (!resultGenerateFiles.isOk()) {
+            return err(resultGenerateFiles.error);
+        }
+        const generatedFileFilepaths: Map<string, GeneratedFileFilepaths> = resultGenerateFiles.value;
+        const generatedFileAbsFilepathsOnService: Map<string, string> = new Map();
+        for (let [fileId, initializingFunc] of containerCreationConfig.getFileGeneratingFuncs().entries()) {
+
+            if (!generatedFileFilepaths.has(fileId)) {
+                return err(new Error("Needed to initialize file for file ID " + fileId +  ", but no generated file filepaths were " +
+                "found for that file ID; this is a Kurtosis bug"));
+            }
+            const filepaths: GeneratedFileFilepaths = generatedFileFilepaths[fileId];
+
+            const promiseFp: Promise<fsPromises.FileHandle> = fsPromises.open(filepaths.getAbsoluteFilepathHere(), 'r'); //TODO - is 'r' the correct mode?; shouldn't I be closing the file (like `defer fp.close()`)
+            promiseFp.catch(error => { 
+                return err(error); 
+            })
+            const resultFp: fsPromises.FileHandle = await promiseFp;
+            const fp: number = resultFp.fd;
+
+            var initalizingFuncErr = initializingFunc(fp);
```

Comment:
let's make sure the `initializingFunc`s return `Result<null>`, in accordance with everything else. That way, the check below (which incorrectly only uses `!=` rather than `!==`) can use `!result.isOk()`